jwks/key.go

```diff
@@ -0,0 +1,145 @@
+package jwks
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math/big"
+)
+
+const typeRSA = "RSA"
+
+var supportedKeyTypes = map[string]struct{}{
+	typeRSA: {},
+}
+
+// JWK defines JSON Web Key structure.
+type JWK struct {
```

Comment:
I’m not sure that JWK and DecodeBase64URLToBigInt should be part of the public interface for the jwks package. There’s a chance we might switch to a third-party library to support more JOSE features, and if these functions are public, we won’t be able to remove them without introducing a breaking change.

I also see that JWK is used in the idptest package. For shared primitives used only internally, we typically place them in the internal directory. In my view, the [idputil](https://github.com/acronis/go-authkit/tree/main/internal/idputil) package could be a good place for them.

Reply:
I considered saving some code in the internal component, but you're right. Let's not give false hope to users of the lib.